### PR TITLE
Add domain events for snippets

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no yes
| New feature? | no yes
| BC breaks? | no yes
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR add the domain events to snippets

### CODE REVIEW

File: src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+     *
+     * @see SecuritySubscriber::handlePersistCreate()
+     */
+    private function isNewNode(NodeInterface $node): bool
+    {
+        /** @var \Countable $properties */
+        $properties = $node->getProperties(
+            $this->propertyEncoder->encode(
+                'system_localized',
+                self::TITLE_FIELD,
```
Comment (niklasnatter) [inline src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php:325]: is there a reason for not using `StructureSubscriber::STRUCTURE_TYPE_FIELD` like for pages? going to trust you here 😇

File: src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+        $data['ext'] = $extensionData;
+
+        return $data;
+    }
+
+    /**
+     * @param NodeInterface<mixed> $node
+     *
+     * @see SecuritySubscriber::handlePersistCreate()
```
Comment (niklasnatter) [inline src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php]: this is not accurate anymore if we are using `self::TITLE_FIELD`

File: src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+use Sulu\Component\DocumentManager\Event\RemoveLocaleEvent;
+use Sulu\Component\DocumentManager\Events;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ */
+class DomainEventSubscriber implements EventSubscriberInterface
```
Comment (niklasnatter) [inline src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php:40]: would you mind to add some checks in the `SnippetControllerTest` like here?
https://github.com/sulu/sulu/pull/5988/files#diff-e053f115525b3f9f06c8c92b827a46c8b284508dd1f7fb45170642e1cd689171

would be nice to have a functional test case for phpcr events 🙂

File: src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php (hunk 1 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php:325]: the structure field is not translated for the articles and the title is also a required field :)

**Pull request merged**